presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -476,4 +537,40 @@ private static void checkInputs(
     {
         return scala.reflect.ClassTag$.MODULE$.apply(clazz);
     }
+
+    private static class SparkPartition
+            implements Comparable<SparkPartition>
+    {
+        private final int partitionId;
+        private long availableCapacityInBytes;
+
+        public SparkPartition(int partitionId, long availableCapacityInBytes)
+        {
+            this.partitionId = partitionId;
+            this.availableCapacityInBytes = availableCapacityInBytes;
+        }
+
+        @Override
+        public int compareTo(SparkPartition o)
+        {
+            return availableCapacityInBytes == o.availableCapacityInBytes ?
```

Comment:
why not simply `this. availableCapacityInBytes  - that. availableCapacityInBytes `? (or the vice versa, depending on the order you are trying to achieve)

Reply:
Nevermind. DIdn't realize we are dealing with a long.